Add Nui, a UI for NATS

Mrxx99 opened the pull request:
## Description

Adds the NATS managment UI called Nui. It seems to be the only managment UI for NATS that is actively maintained (at least what I could find).
https://github.com/nats-nui/nui
https://natsnui.app/

The implementation is inspired  by PgAdmin/PgWeb and RedisInsights implementation.

## Checklist

- Is this feature complete?
  - [x] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [x] Yes
  - [ ] No
- Did you add public API?
  - [x] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [x] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [x] Yes
      - [ ] No
  - [ ] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change require an update in our Aspire docs?
  - [x] Yes (?)
    - Link to aspire-docs issue: todo
  - [ ] No

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/6056)

Mrxx99 committed:
Add Nui a UI for NATS

Mrxx99 committed:
add code examples to XML doc comments

Mrxx99 commented:
this has to updated, after #6259 was merged

danmoseley commented:
@Mrxx99  #6259 has merged now, are you able to update this PR and make un-draft?

Mrxx99 commented:
Yes, will try to finish this this week

danmoseley commented:
@Mrxx99 do you plan to finish this, do you think?

Mrxx99 commented:
Sorry I didn't have much time, but yes I will try to finish it this week finally. I saw now Nui now supports a file initialization which I guess would be preferred over a http call as it is more similar to others using volumes?

Mrxx99 committed:
add password support for Nui

Mrxx99 committed:
updated to latest Nui version 0.6.1

danmoseley commented:
I'm not famliiar with the NATS integration so @eerhardt or @sebastienros will have to answer.

eerhardt committed:
Merge remote-tracking branch 'upstream/main' into feature/nui

eerhardt committed:
PR feedback

- Rename NuiContainerResource to NuiResource
- Use remarks on image tags so the XML docs are correct
- Minor clean up

eerhardt commented on src/Aspire.Hosting.Nats/NatsBuilderExtensions.cs:
```diff
@@ -137,6 +139,155 @@ public static IResourceBuilder<NatsServerResource> WithJetStream(this IResourceB
+                                                 .WithImage(NatsContainerImageTags.NuiImage, NatsContainerImageTags.NuiTag)
+                                                 .WithImageRegistry(NatsContainerImageTags.NuiRegistry)
+                                                 .WithHttpEndpoint(targetPort: 31311, name: "http")
+                                                 .WithVolume(VolumeNameGenerator.Generate(builder, "db"), "/db")
```
Typically we make this "opt-in" with a `.WithDataVolume` extension method on the `IResourceBuilder<NuiResource> builder`.

danmoseley commented:
I'm going to close this as stale @Mrxx99 feel free to reopen if you want to continue with it. thanks!
